### DISCUSSION TITLE
How does init_outputs work? Can it be updated dynamically?

### DISCUSSION THREAD

**Discussion by kevinlinxc** [leon-thomm/Ryven#130] [Q&A] [answered]
Imagine that I didn't know the number of outputs of a node until the user puts in the input. Is it possible to make the number of outputs dynamic?

For a simple example, a node where the user inputs an integer e.g. 5, and the node updates to have 5 outputs.

The actual use case is reading a Pandas dataframe and having each of the columns be available as an output of the node.

**Comment by kevinlinxc** [comment, marked as answer]
Did some digging in the source code and it seems like all I need to do is 
```
self.create_output("Hi")
```
to create a new output.

**Comment by ddevz** [comment]
Neat use case!
But how does it connected to something?  Are you planning to have one "template" output that goes to nodes, which get copied for all new outputs?  Or is the idea to load the pandas frame once during development, then manually connect them?
